doc/tutorials/arrays.md

```diff
@@ -0,0 +1,135 @@
+
+```sh
+# Elektra sorts keys alphabetically
+kdb ls user/tests/parent
+#> user/tests/parent
+#> user/tests/parent/daughter
+#> user/tests/parent/daughter/grandchild
+#> user/tests/parent/son
+
+kdb get user/tests/parent
+#> value
+kdb get user/tests/parent/daughter
+#>
+```
+
+.
+
+## Array Keys
+
+Since Elektra sorts keys alphabetically we can use the key-value pair structure described above to store sequences of values (aka. arrays).
+
+### Empty Arrays
+
+For an **empty array** (`[]`) we just add the [meta-key](../help/elektra-metadata.md) `array`:
+
+```sh
+# Create an empty array with the name `user/tests/sequence`
+kdb setmeta user/tests/sequence array ''
+```
+
+.
+
+### Array Elements
+
+To create an **array element** we start the basename of a key with the `#` character and add the index of the array element afterwards. For example, the commands below adds three elements to our array `user/tests/sequence`:
+
+```sh
+kdb set user/tests/sequence/#0 'First Element'
+kdb set user/tests/sequence/#1 'Second Element'
+# Arrays do not need to be contiguous
+kdb set user/tests/sequence/#3 'Fourth Element'
+```
+
+. As you can see above arrays can contain “empty fields”: The key `user/tests/sequence/#2` is missing.
+
+For array elements with an index larger than `9` we must add **underscores** (`_`) to the basename, so we do not destroy the alphabetic order of the array. For example, to add a eleventh element to our array we use the following command:
+
+```sh
+kdb set user/tests/sequence/#_10 'Eleventh Element'
+```
+
+. The order of the array sequence is still correct afterwards, as the following command shows:
+
+```sh
+# List all array elements
+kdb ls user/tests/sequence/
+#> user/tests/sequence/#0
+#> user/tests/sequence/#1
+#> user/tests/sequence/#3
+#> user/tests/sequence/#_10
+```
+
+. For larger indices we add **one underscore less, than the number of digits** of the index. For example, to add a element with index `1337` (`4` digits) we use the basename `#___1337`. We can also generate the basename programmatically:
+
+```bash
+ruby -e 'print("#", "_" * (ARGV[0].length - 1), ARGV[0])' 12345
+#> #____12345
+
+ruby -e 'print("#", "_" * (ARGV[0].length - 1), ARGV[0])' 0
+#> #0
+
+ruby -e 'print("#", "_" * (ARGV[0].length - 1), ARGV[0])' 42
+#> #_42
+```
+
+.
+
+### Metadata
+
+To make life easier for users, Elektra’s [storage plugins](plugins.md) should save the [basename of the last key in the array parent](../decisions/array.md). Some plugins do not support this feature. In that case you can add this value manually via `kdb setmeta`:
+
+```sh
+# Add array elements
+kdb set user/tests/favorites/superheros/#0 'One-Punch Man'
+kdb set user/tests/favorites/superheros/#1 'Mermaid Man and Barnacle Boy'
+kdb set user/tests/favorites/superheros/#____99999 'The guy with the bow and arrow'
+
+# The metakey `array` should save the basename of the last element.
+kdb setmeta user/tests/favorites/superheros array '#____99999'
```

Comment:
Nice example but the main reason is iteration over all array members.

Reply:
How does this help us to iterate over array elements (, if we support “empty” array elements)? For example, the array above has only three (non-empty) members, while the size of the array is `100000`. I do not think it makes sense to iterate `100000` times in such a case…

Reply:
Seems like we did not fully thought that through.

Especially if arrays are distributed between different namespaces it gets quite tricky.

Reply:
> Seems like we did not fully thought that through.

In that case, can I leave this PR as it is or should I update the text?

Reply:
It is a very good starting point. Thanks again for writing the tutorial!